components/filter/filter.js

```diff
@@ -375,11 +375,11 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 					this._totalAppliedCount--;
 				}
 			} else if (prop === 'values') {
-				shouldRecount = true;
+				if (dimension.searchType !== 'manual' || !dimension.searchValue) shouldRecount = true;
```

Comment:
If there's no search value then we should have all the options and can safely recount.  We want to recount as often as we can in case set values are loaded in asynchronously.  A scenario that shouldn't happen but will break the counts is if after a search, the consumer returns us new values we didn't originally have before searching.  This is easily avoidable though - don't finish loading and pass us the values until you have at least all the keys.  The filter doesn't support paging/load more.